2007-05-24  Aurelien Jarno <aurelien@aurel32.net>

	* elf/Makefile: ignore global registers while looking for
	  undefined symbols. 

--- elf/Makefile.orig
+++ elf/Makefile
@@ -311,7 +311,7 @@
 		  $(filter-out $(map-file),$^) $(load-map-file)		\
 		  -Wl,-soname=$(rtld-installed-name) -T $@.lds
 	rm -f $@.lds
-	nm -u $@ | cmp -s /dev/null -
+	objdump -T $@ | grep 'D. \*UND\*' | cmp -s /dev/null -
 
 # interp.c exists just to get this string into the libraries.
 CFLAGS-interp.c = -D'RUNTIME_LINKER="$(slibdir)/$(rtld-installed-name)"' \
